src/condor_status.V6/status.cpp

```diff
@@ -1204,6 +1203,7 @@ main (int argc, char *argv[])
 		q = collectors->query( * query, rightCallback, rightArg, & errstack );
 		delete collectors;
 	}
+	delete d;
```

Comment:
It looks like this delete needs to be around line 1257, after the last reference to requested_daemon.